src/tagged-type.ts

```diff
@@ -153,17 +157,20 @@ const encodeMap = {
   documentReference: (value: DocumentReference): TaggedRef => ({
     "@ref": { id: value.id, coll: { "@mod": value.coll.name } },
   }),
-  document: (value: Document): TaggedRef => ({
-    "@ref": { id: value.id, coll: { "@mod": value.coll.name } },
+  document: (value: Document): TaggedDoc => ({
+    "@doc": { id: value.id, coll: { "@mod": value.coll.name } },
```

Comment:
I'll follow with this in python, but originally this was intentional. It feels awkward to encode to @doc when it's really a ref (and lacks all of the data and the ts). @ref feels like it communicates more clearly what is going on.

Reply:
In that case, I will change back to encoding Document and NamedDocument to `@ref` and also expect that on round-trip back to the client that `@ref` results in a Document.

Reply:
... unless the document doesn't exist. In that case, it is returned as `@ref` again.

I've reverted the encoding for docs back to using `@ref` and updated tests, including making it two tests, one for doc exists and one for when it doesn't.